kernel/include/timeout_q.h

```diff
@@ -23,6 +23,7 @@ extern "C" {
 static inline void _init_timeout(struct _timeout *t, _timeout_func_t fn)
 {
 	sys_dnode_init(&t->node);
+	t->dticks = 0;
```

Comment:
This... is not the scheme I thought we'd agreed to.  I thought this was simply going to change the convention 

We've had multiple bugs with this kind of separate list handling in the past, because the timer objects may be modified while in that list, and it's not possible to keep execution locked while the callbacks execute.  How does this work if another context aborts a thread while it's queued in the "to be executed" list?  What happens if the callback tries to reschedule itself?

Reply:
It is what I said I was going to do: a solution based on the branch I'd referenced in some issue attached to the problem we were discussing.  Both abort and self-reschedule should work just fine.

It looks like #12499 causes the test I created to pass, so we'll probably go with that.  I'll end up re-opening #12248 though because if you're not going to take the change to managing the timeout list everything else that got added lacks motivation.

Reply:
Please don't dump the whole thing.  The cleaned up list removal workaround has obvious value, as does the test case.

Reply:
I can't add the test case as it will fail until your solution is merged.  The rest is back in #12248 where it started out.